src/components/Header/Header.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import cn from 'classnames';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  setErrorMessage: Dispatch<SetStateAction<Errors>>;
+  onAddTodo: (value: string) => Promise<void>;
+  isInputDisabled: boolean;
+  onToggleAll: () => Promise<void>;
+  todosLength: number;
+  inputRef: React.RefObject<HTMLInputElement> | null;
+  areAllTodosCompleted: boolean;
+};
+
+export const Header: React.FC<Props> = props => {
+  const {
+    setErrorMessage,
+    onAddTodo,
+    isInputDisabled,
+    onToggleAll,
+    todosLength,
+    inputRef,
+    areAllTodosCompleted,
+  } = props;
+  const [inputValue, setInputValue] = useState('');
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (inputValue.trim() === '') {
+      setErrorMessage(Errors.EmptyTitle);
+
+      return;
+    }
+
+    try {
+      await onAddTodo(inputValue.trim());
+      setInputValue('');
+    } catch (err) {}
+  };
+
+  useEffect(() => {
+    inputRef?.current?.focus();
+  }, [todosLength, inputRef]);
+
+  useEffect(() => {
+    if (!isInputDisabled) {
+      inputRef?.current?.focus();
+    }
+  }, [isInputDisabled, inputRef]);
+
+  return (
+    <header className="todoapp__header">
+      {todosLength !== 0 && (
```

Comment:
use this syntax for all the cases(more readable)

```suggestion
      {!!todosLength && (
```